src/dripper/polkadot/PolkadotActions.ts

```diff
@@ -1,223 +1,165 @@
+          `Faucet balance update. It's now ${formatAmount(this.#faucetBalance)} ${networkData.data.currency}`,
+        );
+      }
+    });
   }
 
-  public getFaucetBalance(): bigint | undefined {
+  public async getFaucetBalance(): Promise<bigint> {
+    // This should mean that #faucetBalance is initialized
+    await this.isReady;
+    if (this.#faucetBalance === undefined) {
+      // So this shouldn't be possible
+      throw new Error("#faucetBalance: uninitialized");
+    }
     return this.#faucetBalance;
   }
 
   public async getAccountBalance(address: string): Promise<number> {
-    const { data } = await polkadotApi.query.system.account(address);
+    const balance = await networkData.api.getBalance(address, client);
 
-    const { free: balanceFree } = data;
-
-    return balanceFree
-      .toBn()
-      .div(new BN(10).pow(new BN(networkData.decimals)))
-      .toNumber();
+    return Number(balance / 10n ** BigInt(networkData.data.decimals));
   }
 
   public async isAccountOverBalanceCap(address: string): Promise<boolean> {
-    return (await this.getAccountBalance(address)) > networkData.balanceCap;
+    return (await this.getAccountBalance(address)) > networkData.data.balanceCap;
   }
 
-  async teleportTokens(dripAmount: bigint, address: string, parachain_id: string): Promise<DripResponse> {
-    logger.info("💸 teleporting tokens");
-
-    const dest = {
-      V3: {
-        interior: {
-          X1: {
-            Parachain: parachain_id,
-          },
-        },
-        parents: 0,
-      },
-    };
-
-    const addressHex = polkadotApi.registry.createType("AccountId", address).toHex();
-    const beneficiary = {
-      V3: {
-        interior: {
-          X1: {
-            AccountId32: { id: addressHex, network: null },
-          },
-        },
-        parents: 0,
-      },
-    };
-
-    const assets = {
-      V3: [
-        {
-          fun: { Fungible: dripAmount },
-          id: {
-            Concrete: {
-              interior: "Here",
-              parents: 0,
-            },
-          },
-        },
-      ],
-    };
-
-    const weightLimit = { Unlimited: null };
-
-    const feeAssetItem = 0;
-
-    const transfer = polkadotApi.tx.xcmPallet.limitedTeleportAssets(
-      dest,
-      beneficiary,
-      assets,
-      feeAssetItem,
-      weightLimit,
-    );
-
-    if (!this.account) throw new Error("account not ready");
-    const hash = await transfer.signAndSend(this.account, { nonce: -1 });
-
-    const result: DripResponse = { hash: hash.toHex() };
-    return result;
+  async teleportTokens(dripAmount: bigint, address: string, parachain_id: number): Promise<DripResponse> {
+    logger.info(`💸 teleporting tokens to ${address}, parachain ${parachain_id}`);
+
+    const addressBinary = Binary.fromBytes(encodeAccount(address));
+
+    const tx = await networkData.api.getTeleportTx({
+      dripAmount,
+      address: addressBinary,
+      parachain_id,
+      client,
+      nonce: this.#nonce++,
```

Comment:
Is it possible that:

1. We increase this nonce here.
2. We fail to send the transaction (for example a temporary network issue).
3. Upon a next request, we increase the nonce again.
4. The faucet is bricked, because it tries to send transactions with too high nonce?

Reply:
Hmm, that's a good point actually. Let's update the nonce from the RPC periodically.

Reply:
There's a complication with that idea though. We can only synchronize nonce when there aren't any pending transactions, as the nonce gets updated when transactions are finalized.  
Although, a check for pending transactions inside of `PolkadotActions` shouldn't be hard to do.

Reply:
WDYT? https://github.com/paritytech/polkadot-testnet-faucet/pull/392/commits/7c2d0244f96139f9e11eac43f2caa3f7eeceff6e

Reply:
Looks good to me 👍